compactor2/src/driver.rs

```diff
@@ -237,10 +227,16 @@ async fn try_compact_partition(
 
             // Update the catalog to reflect the newly created files, soft delete the compacted files and
             // update the upgraded files
+            let files_to_delete = match file_classification.files_to_compact_or_split {
+                FilesToCompactOrSplit::FilesToCompact(files) => files,
+                FilesToCompactOrSplit::FilesToSplit(files) => {
+                    files.iter().map(|f| f.file.clone()).collect()
+                }
+            };
```

Comment:
A minor comment is this might be nice to refactor into a function on `FilesToCompactOrSplit`

like 

```suggestion
            let files_to_delete = file_classification.files_to_compact_or_split.files();
```

Which would keep the driver logic more concise